jquery.slideandswipe.js

```diff
@@ -30,141 +30,198 @@
-                return /Safari/.test(navigator.userAgent) && /Apple Computer/.test(navigator.vendor);
-            }
-
-            function isChrome() {
-                return /Chrome/.test(navigator.userAgent) && /Google Inc/.test(navigator.vendor);
-            }
-
-            /**
-             * Manually update the position of the nav on drag
-             */
-            function scrollNav(distance, duration) {
-                nav.css('transition-duration', (duration / 1000).toFixed(1) + 's');
-
-                if(distance >= 0) {
-                    distance = 0;
-                }
-                if(distance <= navWidth) {
-                    distance = navWidth;
-                }
-                if(isSafari() || isChrome()) {
-                   nav.css('-webkit-transform', 'translate(' + distance + 'px,0)');
-                }
-                else{
-                   nav.css('transform', 'translate(' + distance + 'px,0)');
-                }
-                if(distance == '0') {
-                    $('.ssm-toggle-nav').addClass('ssm-nav-visible');
-                    $('html').addClass('is-navOpen');
-                    $('.ssm-overlay').fadeIn();
-                }
-            }
-
-            /**
-             * Open / close by click on burger icon
-             */
-            var hideNavigation = (function() {
-                nav.removeClass('ssm-nav-visible');
-                scrollNav(navWidth, settings.speed);
-                $('html').removeClass('is-navOpen');
-                $('.ssm-overlay').fadeOut();
-            });
-
-            var showNavigation = (function() {
-                nav.addClass('ssm-nav-visible');
-                scrollNav(0, settings.speed);
-            });
-
-            $('.ssm-toggle-nav').click(function(e) {
-                if(nav.hasClass('ssm-nav-visible')) {
-                    hideNavigation();
-                }
-                else{
-                    showNavigation();
-                }
-                e.preventDefault();
-            });
-        }
-    ;
+	'use strict';
+
+	var slideAndSwipe =
+
+		$.fn.slideAndSwipe = function(options) {
+
+			var nav = $(this); // get the element to swipe
+			var navWidth = -nav.outerWidth();
+			var transInitial = navWidth;
+
+			if( "object" === typeof options && "boolean" === typeof options.hide ){
+				if( options.hide && "undefined" !== typeof hideNavigation ){
+					hideNavigation();
+				}
+			}
+
+			if( "object" === typeof options && "boolean" === typeof options.show ){
+				if( options.show || options.show && ( "boolean" === typeof options.hide && ! options.hide ) ){
+					showNavigation();
+				}
+			}
+
+			// get settings
+			var settings = $.extend({
+				triggerOnTouchEnd   : true,
+				swipeStatus         : swipeStatus,
+				allowPageScroll     : 'vertical',
+				threshold           : 100,
+				excludedElements    : 'label, button, input, select, textarea, .noSwipe',
+				speed               : 500

```

Comment:
would you set the default speed to  `250` as it was in the master branch?

Reply:
@JoanClaret I tested in smartphone and best speed is 500, but I going edit to 250